src/Kunstmaan/AdminBundle/Helper/AdminRouteHelper.php

```diff
@@ -17,18 +19,18 @@ class AdminRouteHelper
     protected $adminKey;
 
     /**
-     * @var SlugRouter $slugRouter
+     * @var Request $request
      */
-    protected $slugRouter;
+    protected $request;
 
     /**
      * @param string $adminKey
-     * @param SlugRouter $slugRouter
+     * @param RequestStack $requestStack
      */
-    public function __construct($adminKey, SlugRouter $slugRouter)
+    public function __construct($adminKey, RequestStack $requestStack)
     {
-        $this->adminKey = $this->normalizeUrlSlice($adminKey);
-        $this->slugRouter = $slugRouter;
+        $this->adminKey = $adminKey;
+        $this->request = $requestStack->getCurrentRequest();
```

Comment:
There could be more than one request during the lifetime of this service (subrequests, forwards, etc). It’s better to just store the `requestStack` and fetch the request each time.